src/services/mock/generate

```diff
@@ -1,26 +1,29 @@
 #!/bin/bash
 
+if ! sed --version 2>/dev/null | grep -q 'GNU' 2>/dev/null; then
+    echo -e '** GNU sed required **\n' >&2
+    exit 1
+fi
 
+cd "$(dirname "$0")"
 CHECK_FILE='checkpoint.js'
 
-
 cylc reg "one" one
-
-
 if cylc run "one" --fcp=20000102T0000Z --no-detach; then
     cp \
         "${HOME}/cylc-run/one/work/20000102T0000Z/checkpoint/checkpoint" \
         "${CHECK_FILE}"
     rm -r "${HOME}/cylc-run/one"
 else
+    echo -e '** Suite run failed **\n' >&2
     exit 1
 fi
 
-# This line goes to the top of the file, disabling eslint for the checkpoint.js file
-sed -i "1 i\/* eslint-disable */\n" "${CHECK_FILE}"
-# This one goes to the second line, prepending the JS variable declaration before the JSON string
-sed -i "3 s/^/const checkpoint = /" "${CHECK_FILE}"
-# Replace globally the user running the workflow
-sed -i "s/${USER}/cylc/g" "${CHECK_FILE}"
-# Finally we export it so we can use it from JS modules
-echo -e '\nexport { checkpoint }' >> "${CHECK_FILE}"
+sed -i \
+    -e "1s|^|const checkpoint = |" \
+    -e "1s|^|/* eslint-disable */\n|" \
+    -e "s|$USER|user|g" \
+    -e "s|$(hostname)|localhost|g" \
+    -e "s|$(hostname | tr '[:upper:]' '[:lower:]')|localhost|g" \
+    -e "\$s|$|\nexport { checkpoint }|" \
+    "${CHECK_FILE}"
```

Comment:
:rocket: nice! Didn't think in combining everything. Hopefully will remember it for next time, but easier to read.